### PR TITLE
Milestone #2 Complete:Components and Redux store

### PR DESCRIPTION
**Pull request by Elbie-em**
This branch consists of all of the necessary requirements specified [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/).

* Prepared the directory structure
* Implemented the App component: Made it responsible for rendering the BooksFrom and BookList components
* Implemented the BooksList component
* Implemented the Book component
* Implemented the BooksForm component
* Defined the Redux actions and reducers
* Initialized the Redux store
* Deployed Updated version to Heroku

You can see the updated deployed version by clicking [mbbookstore.herokuapp.com](https://mbbookstore.herokuapp.com/)

**Note: I am working on this project solo as my partner and I were unpaired by student success, I was advised to proceed on my own.**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './App';
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
+import App from './components/App';
+import rootReducer from './reducers/index';
+
+const reduxStore = createStore(rootReducer);
```
Comment (wathigo) [inline src/index.js:8]: **REQUIRED**
* It adds example books to initial state object. [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/InitialBooks.png).
See the [documentation](https://redux.js.org/api/createstore) on how to create store with a pre-loaded state using createStore.

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required :heavy_multiplication_x: 
Great job on this milestone.
Let's fix one issue before we proceed to the next milestone.

See the comment :point_down: 

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo.**

File: src/index.js (hunk 1 above)
Comment (Elbie-em) [reply src/index.js:8]: As per coding conventions and the majority of react-redux tutorials I have come across, the best-suited place for this initial state object is in the file you have the reducer as the main index.js file shouldn't contain inner core responsibilities or exports as it is not good practice. Please advise me further if anything
Screenshot 2020-10-23 at 09 58 18
.

**Review by mosaaleb — APPROVED** [review]
#### Milestone Approved :100: 

Hi @Elbie-em, I agree that polluting components with pre-populated data that can be encapsulated elsewhere may be considered as an anti-pattern.

You are good to go :+1: 

##### Muhammad Ebeid

**Pull request merged**